velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -682,5 +682,38 @@ TEST_F(StringTest, translateNonconstantMatch) {
   expected = makeFlatVector<std::string>({"åbaæçè", "åæcèaç"});
   testTranslate({input, match, replace}, expected);
 }
+
+TEST_F(StringTest, url_decode) {
+  auto UrlDecodeSimpleTest = [&](const std::optional<std::string>& input,
+                                 const std::string& expected) {
+    EXPECT_EQ(
+        evaluateOnce<std::string>("url_decode(c0)", input).value(), expected);
+  };
+  UrlDecodeSimpleTest("AaBbC", "AaBbC");
+  UrlDecodeSimpleTest("A+B+C", "A B C");
+  UrlDecodeSimpleTest("AaBbC%23", "AaBbC#");
+  UrlDecodeSimpleTest("AaBbC%24%5A", "AaBbC$Z");
+  UrlDecodeSimpleTest("AaBbC%25FF", "AaBbC%FF");
+  // Illegal case.
+  EXPECT_THROW(
+      evaluateOnce<std::string>("url_decode(c0)", std::optional{"AaBbC%5"}),
+      VeloxUserError);
+  EXPECT_THROW(
+      evaluateOnce<std::string>("url_decode(c0)", std::optional{"Aa%6HBbC"}),
+      VeloxUserError);
+
+  auto rowType = ROW({{"c0", VARCHAR()}});
+  auto exprSet = compileExpression("url_decode(c0)", rowType);
+  auto testUrlDecode = [&](const VectorPtr& input, const auto& expected) {
+    auto result = evaluate(*exprSet, makeRowVector({input}));
+    velox::test::assertEqualVectors(expected, result);
+  };
+  auto input = makeFlatVector<std::string>(
```

Comment:
Why not use `UrlDecodeSimpleTest`?

Reply:
Just to have a test on batch input. Thanks!

Reply:
We don't need batch here, evaluateOnce will makeRowVector

Reply:
Just fixed. Indeed, it looks unnecessary to evaluate multiple input cases together. Just re-used the simple test in latest commit. Thanks!